rmw_fastrtps_dynamic_cpp/src/rmw_client.cpp

```diff
@@ -97,9 +97,11 @@ rmw_create_client(
   const rosidl_service_type_support_t * type_support = get_service_typesupport_handle(
     type_supports, rosidl_typesupport_introspection_c__identifier);
   if (!type_support) {
+    rcutils_reset_error();
     type_support = get_service_typesupport_handle(
       type_supports, rosidl_typesupport_introspection_cpp::typesupport_identifier);
     if (!type_support) {
+      rcutils_reset_error();
```

Comment:
Why are we not getting old messages here, as it is done for `rmw_fastrtps_cpp` ?

Reply:
Oops, I missed that one. Thanks @MiguelCompany !

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/495/commits/b45a8ec945452d1a78a41dfac9fdc40b239b191b.